clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4132,15 +4132,15 @@ void Sema::finalizeSYCLDelayedAnalysis(const FunctionDecl *Caller,
   }
 }
 
-bool Sema::checkAllowedSYCLInitializer(VarDecl *VD, bool CheckValueDependent) {
+bool Sema::checkAllowedSYCLInitializer(VarDecl *VD) {
   assert(getLangOpts().SYCLIsDevice &&
          "Should only be called during SYCL compilation");
 
   if (VD->isInvalidDecl() || !VD->hasInit() || !VD->hasGlobalStorage())
     return true;
 
   const Expr *Init = VD->getInit();
-  bool ValueDependent = CheckValueDependent && Init->isValueDependent();
```

Comment:
I don't understand why this was done in the first place. Is there anything in the commit message of the change where `CheckValueDependent `was introduced,  which explains why this was done?

Reply:
@premanandrao had introduced `CheckValueDependent` in this [PR](https://github.com/intel/llvm/pull/2019)
I did check with him yesterday as to why it was added/needed. His response:

> I believe that this was added because of the various places that this routine is called from. When called before instantiations had completed, we needed to check this to avoid the cases where it might be dependent. On the later calls, the instantiations were expected to be complete, so the check wasn't needed. (This is what I am gleaning from the PR you linked.)